src/csharp/Grpc.Core.Api/DeserializationContext.cs

```diff
@@ -39,13 +39,30 @@ public abstract class DeserializationContext
         /// Also, allocating a new buffer each time can put excessive pressure on GC, especially if
         /// the payload is more than 86700 bytes large (which means the newly allocated buffer will be placed in LOH,
         /// and LOH object can only be garbage collected via a full ("stop the world") GC run).
-        /// NOTE: Deserializers are expected not to call this method more than once per received message
+        /// NOTE: Deserializers are expected not to call this method (or other payload accessor methods) more than once per received message
         /// (as there is no practical reason for doing so) and <c>DeserializationContext</c> implementations are free to assume so.
         /// </summary>
         /// <returns>byte array containing the entire payload.</returns>
         public virtual byte[] PayloadAsNewBuffer()
```

Comment:
For other platforms than `netstandard2.0`, one still needs to use this old method, which is inefficient and very GC heavy for large messages. With the machinery added in this PR, it would be very easy to add a new overload that still copies the data, but allows populating a buffer provided by the user (which can eliminate the GC overhead) - leaving that for followup PRs.